core/traffic_class.h

```diff
@@ -124,13 +124,9 @@ class TrafficClass {
 
   virtual void Traverse(TraverseTcFn f, void *arg) const { f(this, arg); }
 
-  size_t Size() const {
-    size_t sz = 0;
-    Traverse([](const TrafficClass *,
-                void *arg) { *reinterpret_cast<size_t *>(arg) += 1; },
-             static_cast<void *>(&sz));
-    return sz;
-  }
+  // Returns the number of TCs in the TC subtree rooted at this, including this
+  // TC.  Returns 0 upon error.
```

Comment:
Is it ever possible (in normal situations) to return 0? If not, shouldn't it be CHECK() or DCHECK()?